eng/CodeAnalysis.ruleset

```diff
@@ -64,7 +64,6 @@
       <Rule Id="CA1308" Action="None" />             <!-- Normalize strings to uppercase -->
       <Rule Id="CA1309" Action="None" />             <!-- Use ordinal stringcomparison -->
       <Rule Id="CA1401" Action="Warning" />          <!-- P/Invokes should not be visible -->
-      <Rule Id="CA1416" Action="None" />             <!-- Validate platform compatibility; https://github.com/dotnet/runtime/issues/41354 -->
```

Comment:
Please change the Action rather than deleting it.  We've been trying to have this contain a complete list of all rules.

Reply:
Per [this comment](https://github.com/dotnet/runtime/issues/41354#issuecomment-682142247), please actually leave the analyzer disabled in this PR so that we can cleanly port this PR's merge comment into 5.0.0 without enabling the analyzer in the 5.0 branch (we want to fix the errors, but not enable it there).

There will need to be a separate PR to enable the analyzer, which we won't merge to 5.0.0.

Reply:
Based on the decision that we don't need to port this into 5.0, we can change course here. Let's set the analyzer to `Warning`.

Reply:
Fixed.